spec/jobs/scrape_keyword_job_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ScrapeKeywordJob do
+  include ActiveJob::TestHelper
+
+  describe '#perform' do
+    context 'given an ERROR occurs' do
+      context "when the keyword doesn't exist" do
+        it 'DOES NOT retry' do
+          perform_enqueued_jobs do
+            described_class.perform_later(keyword_id: 42)
+          end
+
+          assert_performed_jobs 1
+        end
+      end
+
+      context 'when an unexpected error occurs' do
+        it 'retries at most 3 times' do
+          keyword = Fabricate(:keyword)
+
+          stub = Google::ScrapeService.new(keyword_id: keyword.id)
```

Comment:
This syntax can work, but the team usually uses the following to set up stubs/mocks:

```suggestion
          scrape_service = instance_double (Google::ScrapeService)
```

Reply:
Fixed in 57151f445a6060b23ed73c52de53e51af407d0b9